modules/png-extraction/ImageExtractor.py

```diff
@@ -131,6 +134,14 @@ def get_tuples(plan, outlist = None, key = ""):
                     value = str(value)
                 outlist.append((key + aa, value))
                 # appends name, value pair for this file. these are later concatenated to the dataframe
+    # appends the private tags
```

Comment:
Looks neat. But I have not tested yet.

Once you fix the other issues and submit a fixed pull request, I will check it for functionality.